rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,763 @@
+          description: OK
+        '401':
+          description: Unauthorized / Invalid Credentials Provided
+
+  # Could also consider /v1/tables/{identifier}
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      # Consider moving this to query parameters,so it can be more easily URL encoded
+      # in case of dots or special characters
+      - name: namespace
+        in: path
+        description: Namespace the table is in
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "prod"
+          multipart_namespace:
+            value: "prod.accounting"
+      - name: table
+        in: path
+        description: Name of the table to load
+        required: true
+        schema:
+          type: string
+        example: "sales"
+    get:
+      tags:
+        - Catalog API
+      summary: Load a given table from a given namespace
+      operationId: loadTable
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetTableResponse'
+        '401':
+          description: Unauthorized
+        # Using 412, `Precondition Failed`, instead of 404, as 404 makes monitoring response codes from ELBs
+        #  very difficult - Hard to tell if clients or servers are misconfigured and calling non-existent routes
+        #  or missing routes versus expected error cases such as NoSuchTableException (expected meaning that
+        #  a person who is on call shouldn't be paged for this but 404 they might need to be).
+        '412':
+          description: NoSuchTableException
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
+    put:
+      tags:
+        - Catalog API
+      summary: Commit an in progress create (or replace) table transaction
+      operationId: commitTable
+      description: Commit a pending create (or replace) table transaction, e.g. for doCommit.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CommitTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CommitTableResponse'
+        '401':
+          description: Unauthorized / Invalid Credentials
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergResponseObject'
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog, optionally purging the underlying data
+      operationId: dropTable
+      description: Remove a table from the catalog, optionally dropping the underlying data
+      parameters:
+        - name: purge
```

Comment:
Might want to change this to `purgeRequested`. I don't think there is a requirement to purge, but we do want to pass it if the engine did.

Reply:
Yeah that makes sense.

The doc comment on the API does somewhat imply that `purge` is somewhat optional, and there are a lot of use cases where somebody who implements things on their server side might choose to reject `purge` or ignore it.

Reply:
The purge option also changes the semantics of the API, as drop could be synchronous and purge might make the API asynchronous and return 202 instead of 200.

Reply:
I'll add 202 to the response as well.